metadata_store_scripts/V2025.12.17.0__update_generate_masking_parameters_procedure.sql

```diff
@@ -0,0 +1,454 @@
+                        [condition_alias] NVARCHAR(255) '$.alias',
+                        [algorithm] NVARCHAR(255) '$.algorithm'
+                    ) AS cond
+                WHERE
+                    r.assigned_algorithm IS NOT NULL
+                    AND r.assigned_algorithm <> ''
+                    AND ISJSON(r.assigned_algorithm) = 1
+                    AND JSON_VALUE(r.assigned_algorithm, '$.key_column') IS NOT NULL
+                    AND cond.[condition_alias] IS NOT NULL
+                    AND cond.[condition_alias] = @filter_alias
+            ) AS filter_check
+            WHERE
+                r.dataset = @dataset
+                AND r.specified_database = @specified_database
+                AND r.specified_schema = @specified_schema
+                AND r.identified_table = @identified_table
+                AND r.assigned_algorithm IS NOT NULL
+                AND r.assigned_algorithm <> ''
+                AND NOT (
+                    ISJSON(r.assigned_algorithm) = 1
+                    AND (LEFT(LTRIM(r.assigned_algorithm), 1) = '[')
+                )
+                AND (
+                    @filter_alias = ''
+                    OR filter_check.filter_found = 1
+                )
+        )
+        BEGIN
+            RAISERROR (
+                @msg,
+                16, 1, -- Severity and State
+                @dataset,
+                @specified_database,
+                @specified_schema,
+                @identified_table,
+                @filter_alias_display
+            );
+            RETURN;
+        END;
+
+    -- base_ruleset_filter - Filter to target table, add ADF type mappings, 
+    -- compute metadata, and exclude key columns
+    WITH base_ruleset_filter AS (
+        SELECT
+            r.dataset,
+            r.specified_database,
+            r.specified_schema,
+            r.identified_table,
+            r.identified_column,
+            r.identified_column_type,
+            r.identified_column_max_length,
+            r.row_count,
+            r.ordinal_position,
+            r.assigned_algorithm,
+            r.algorithm_metadata,
+            t.adf_type,
+            CONCAT(
+                'x',
+                CONVERT(
+                    VARCHAR(MAX),
+                    CONVERT(VARBINARY, r.identified_column),
+                    2
+                )
+            ) AS encoded_column_name,
+            -- algorithm_metadata is always expected to be a JSON object (or NULL), 
+            -- so JSON_VALUE is safe without ISJSON/CASE
+            JSON_VALUE(r.algorithm_metadata, '$.date_format') AS [date_format],
+            CONVERT(BIT, JSON_VALUE(r.algorithm_metadata, '$.treat_as_string')) AS treat_as_string,
+            -- Add column width estimate here
+            CASE
+                WHEN r.identified_column_max_length > 0 THEN r.identified_column_max_length + 4
+                ELSE @column_width_estimate
+            END AS column_width_estimate
```

Comment:
note: you're adding a regression here - what happened to the capped identified column width?